strawberryfields/io.py

```diff
@@ -244,6 +245,66 @@ def is_free_param(param):
     return prog
 
 
+def serialize_program(prog, eng=None):
+    """Converts a Strawberry Fields program into valid Strawberry Fields code.
+
+    Args:
+        prog (Program): the Strawberry Fields program
+        eng (Engine): the Strawberryfields engine
```

Comment:
Oh, this is a good idea to have the engine as input! I can't remember if the server version was already like this haha

Reply:
Nope, it didn't. I actually wrote this one from scratch since I thought that was easier. I like being able to pass an engine as well. It makes it possible to generate code using different engines and programs (and if skipping the engine, it'll just return the circuit code).

Reply:
More on the visual side at a later stage, maybe this could also be a drop-down option? That way we wouldn't have to worry about input validation (e.g., making sure that the user did indeed input the correct engine).

Reply:
@antalszava how do you mean exactly?

Reply:
The `eng` argument that is passed to `generate_code` is given as user input. This in itself would then require input validation, so that there is some sort of an error raised if the wrong value is inputted.

My thinking previously was that that can be okay when embedded into a webpage because we could then create a drop-down menu where the possible options are pre-defined (hence there's no need for input validation). For other use cases maybe it's worth including some sort of validation or defining the accepted type of engines here in advance.